### PR TITLE
[libaaplus] Add new port

### PR DESCRIPTION
**Pull request by Matioupi**
This is a PR proposal for a new port of AA+ library by naughter software :
http://www.naughter.com/aa.html

I checked with the lib author that he is OK with having it integrated into vcpkg.

### CODE REVIEW

File: ports/libaaplus/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,300 @@
+PROJECT (AstronomicalAlgorithms)
+
+cmake_minimum_required(VERSION 3.11)
+cmake_policy(VERSION 3.11)
+
+project(libaaplus LANGUAGES CXX)
+
+set(LIBS_SRC
```
Comment (JackBoosY) [inline ports/libaaplus/CMakeLists.txt]: If all source files need to be added, we prefer to use:
`file(GLOB LIBS_SRC ${CMAKE_CURRENT_LIST_DIR}/*.cpp)`

File: ports/libaaplus/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+  AAVSOP87_MAR.cpp
+  AAVSOP87_MER.cpp
+  AAVSOP87_NEP.cpp
+  AAVSOP87_SAT.cpp
+  AAVSOP87_URA.cpp
+  AAVSOP87_VEN.cpp
+  stdafx.cpp
+)
+
+set(PUBLIC_HEADERS
```
Comment (JackBoosY) [inline ports/libaaplus/CMakeLists.txt]: If all source files need to be added, we prefer to use:
`file(GLOB PUBLIC_HEADERS ${CMAKE_CURRENT_LIST_DIR}/*.h)`

File: ports/libaaplus/CONTROL [hunk 3]
```diff
@@ -0,0 +1,3 @@
+Source: libaaplus
+Version: 2.1.0
+Description: libaaplus is an astronomical computations library by naughter software : http://www.naughter.com/aa.html
```
Comment (JackBoosY) [inline ports/libaaplus/CONTROL]: Can you add homepage here?

File: ports/libaaplus/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,35 @@
+include(vcpkg_common_functions)
```
Comment (JackBoosY) [inline ports/libaaplus/portfile.cmake]: We no longer need to include this, please remove this line.

File: ports/libaaplus/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+vcpkg_install_cmake()
+vcpkg_copy_pdbs()
+
+file(
+    INSTALL ${SOURCE_PATH}/AA+.htm
+    DESTINATION ${CURRENT_PACKAGES_DIR}/share/libaaplus
+    RENAME copyright
+)
+
+vcpkg_test_cmake(PACKAGE_NAME LIBAAPLUS MODULE)
```
Comment (JackBoosY) [inline ports/libaaplus/portfile.cmake]: vcpkg_test_cmake is deprecated, please remove this line.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/libaaplus/CMakeLists.txt (hunk 1 above)
Comment (Matioupi) [reply ports/libaaplus/CMakeLists.txt]: done. I add all and substracted the one AATest.cpp file which is used to build the test executable, I hope this fit the rules.

File: ports/libaaplus/CMakeLists.txt (hunk 2 above)
Comment (Matioupi) [reply ports/libaaplus/CMakeLists.txt]: done

File: ports/libaaplus/CONTROL (hunk 3 above)
Comment (Matioupi) [reply ports/libaaplus/CONTROL]: done

File: ports/libaaplus/portfile.cmake (hunk 4 above)
Comment (Matioupi) [reply ports/libaaplus/portfile.cmake]: done

File: ports/libaaplus/portfile.cmake (hunk 5 above)
Comment (Matioupi) [reply ports/libaaplus/portfile.cmake]: done

**Review by JackBoosY — APPROVED** [review]

**Comment by Matioupi** [comment]
I don't understand why the vcpkg-Linux-PR #20687 failed... will need a learning curve here and help from gurus !

**Comment by JackBoosY** [comment]
@Matioupi It's a fake error, ignore it.

**Comment by dan-shaw** [comment]
@Matioupi LGTM, the MacOS CI pipeline is down now, so we'll have to wait a bit before we can merge this.

**Review by dan-shaw — APPROVED** [review]

**Comment by Matioupi** [comment]
> 
> 
> @Matioupi LGTM, the MacOS CI pipeline is down now, so we'll have to wait a bit before we can merge this.

Seems like all the pipelines have issue this times ! One day the windows test pass, and next day it fails without obvious reasons.

**Comment by JackBoosY** [comment]
@Matioupi Yes, we should update baseline in master branch.

**Comment by Matioupi** [comment]
is this long CI test queue time normal ? I see that the pipelines have restarted for some other PR tests.

**Comment by JackBoosY** [comment]
After the CI system is fixed, we will merge it.

**Comment by Matioupi** [comment]
Hello, is the CI system still down ?

**Pull request merged**